rpc/src/v1/impls/eth_pubsub.rs

```diff
@@ -220,18 +218,10 @@ impl<C: LightClient> LightChainNotify for ChainNotificationHandler<C> {
 }
 
 impl<C: BlockChainClient> ChainNotify for ChainNotificationHandler<C> {
-	fn new_blocks(
-		&self,
-		_imported: Vec<H256>,
-		_invalid: Vec<H256>,
-		route: ChainRoute,
-		_sealed: Vec<H256>,
-		// Block bytes.
-		_proposed: Vec<Bytes>,
-		_duration: Duration,
-	) {
+	fn new_blocks(&self, new_blocks: NewBlocks) {
+		if self.heads_subscribers.read().len() == 0 && self.logs_subscribers.read().len() == 0 { return }
```

Comment:
`is_empty()` instead of `len() == 0`? Or maybe if it's a hot code consider storing `AtomicBool` intead of acquring those two locks

Reply:
I think this is OK to leave as a `read()`, the only time we need a write lock is when adding or removing subscribers. Updated to use `is_empty()`